File: src/py/exceptions.py
def DoStuff(inError):
    if inError:
        raise Exception("An error occured!")
    
    print("Doing stuff!")

try:
    DoStuff(False)
    DoStuff(True)
    DoStuff(False)
except Exception as ex:
    print("Error: " + ex.args[0])

File: src/cpp/exceptions.cpp
#include <exception>
#include <iostream>

void DoStuff(bool inError)
{
    if (inError)
    {
        throw std::runtime_error("An error occured!");
    }
    std::cout << "Doing stuff!" << std::endl;
}

int main()
{
    try
    {
        DoStuff(false);
        DoStuff(true);
        DoStuff(false); // not reached
    }
    catch (std::exception &ex)
    {
        std::cout << "ERROR: " << ex.what();
    }

    return 0;
}